src/transformers/models/gemma2/modeling_gemma2.py

```diff
@@ -840,6 +822,11 @@ def forward(
                 dtype=inputs_embeds.dtype,
             )
 
+        if cache_position is None:
```

Comment:
okey, this should always work actually since the seq length gets `layer_idx=0`. Just one question, isn't it a bit misleading if some layers will have `get_seq_length()` number of tokens while others no more than sliding window length?

Reply:
@zucchini-nlp yes, if `get_seq_length` gets called on the wrong layer we will have problems! I'm going to add an exception if it gets called on layer_idx != 0 (I doubt we need it).

Reply:
okey sounds good, as long as the function of `get_seq_length` is transparent for users, to reduce number of cache-related question we get 😄